app/services/FileProcessingService.scala

```diff
@@ -39,82 +42,89 @@ import scala.concurrent.ExecutionContext.Implicits.global
-        throw new ERSFileProcessingException(("callback data storage in sessioncache failed "), "Exception storing callback data")
-    } .recover {
-      case e:Throwable => Logger.error(s"storeCallbackData failed with Exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
+        throw ERSFileProcessingException(("callback data storage in sessioncache failed "), "Exception storing callback data")
+    }.recover {
+      case e: Throwable => Logger.error(s"storeCallbackData failed with Exception ${e.getMessage}, timestamp: ${System.currentTimeMillis()}.")
         throw e
     }
+
     Logger.warn(s"Total rows for schemeRef ${schemeInfo.schemeRef}: $totalRows")
     auditEvents.totalRows(totalRows, schemeInfo)
     res1
   }
 
-  def processCsvFile(callbackData: CallbackData, empRef: String)(implicit hc: HeaderCarrier, schemeInfo: SchemeInfo,request: Request[_])={
+  def processCsvFile(callbackData: UpscanCallback, empRef: String)(implicit hc: HeaderCarrier, schemeInfo: SchemeInfo, request: Request[_]) = {
     val startTime = System.currentTimeMillis()
-    readCSVFile(callbackData).map { fileData =>
+    readCSVFile(callbackData.downloadUrl).map { fileData =>
       Logger.info(" 2. Invoke Data generator ")
       deliverBESMetrics(startTime)
 
-      val sheetName = callbackData.name.getOrElse("").replace(".csv","")
+      val sheetName = callbackData.name.replace(".csv","")
       val result: ListBuffer[Seq[String]] = getCsvData(fileData)(schemeInfo, sheetName,hc,request)
       val schemeData: SchemeData = SchemeData(schemeInfo, sheetName, None, result)
 
       Logger.info("2.1 result contains: " + result)
       Logger.debug("No if SchemeData Objects " + result.size)
-      (sendScheme(schemeData, callbackData, empRef), schemeData.data.size)
+      (sendScheme(schemeData, empRef), schemeData.data.size)
     }
   }
 
-  def readFile(collection: String, id: String) : Iterator[String] = {
-    Logger.debug("1. Read File ")
-    try {
-      val stream = ersConnector.readAttachmentUri(collection, id)
-      new StaxProcessor(stream)
-    }
-    catch {
-      case _: Throwable => throw new ERSFileProcessingException(Messages("ers.exceptions.fileProcessingService.failedStream"), Messages("ers.exceptions.fileProcessingService.stax"))
+  private[services] def readFile(downloadUrl: String): Iterator[String] = {
+    val stream = ersConnector.upscanFileStream(downloadUrl)
+    val targetFileName = "content.xml"
+    val zipInputStream = new ZipInputStream(stream)
+    @scala.annotation.tailrec
+    def findFileInZip(stream: ZipInputStream): InputStream = {
+      Option(stream.getNextEntry) match {
+        case Some(entry) if entry.getName == targetFileName =>
+          stream
+        case Some(a) =>
+          findFileInZip(stream)
+        case None =>
+          throw ERSFileProcessingException(
+            Messages("ers.exceptions.fileProcessingService.failedStream"),
```

Comment:
The implicit in scope comes from the static application. The messages object uses the implicit messages from the import scope. I believe this to be correct for the current pattern of the service. It would be changed if it were DI'd or play upgraded.